eng/scripts/tests/Invoke-GenerateAndBuildV2.tests.ps1

```diff
@@ -0,0 +1,52 @@
+#Requires -Version 7.0
+. (Join-Path $PSScriptRoot ".." ".." "common" "scripts" "Helpers" PSModule-Helpers.ps1)
+
+Install-ModuleIfNotInstalled "Pester" "5.3.3" | Import-Module
+
+Describe "Generate And Build SDK" -Tag "pipeline" {
+    BeforeAll {
+        # . $PSScriptRoot/../Invoke-GenerateAndBuildV2.ps1
+        # clean
+        $sdkRootPath =  (Join-Path $PSScriptRoot .. .. ..)
+        $sdkFolder = (Join-Path $sdkRootPath "sdk" "deviceupdate" "Azure.IoT.DeviceUpdate")
+        if (Test-Path $sdkFolder) {
```

Comment:
Is this removing files that are already checked in? If so, maybe it would be better to copy them into a test directory and test against that target instead?

Reply:
Thanks @benbp . It is better not to remove the checkedin files. But we cannot copy them to a separate test directory, because all the sdk directory is under sdk\{serviceDirectory}, and the {serviceDiretory} is retried from the readme.md path. And we will use the {serviceDirectory} to get the swagger files in azure-rest-api-specs repo, so we cannot mock the readme.md path.